File: Storage/Code/242.有效的字母异位词.py
#
# @lc app=leetcode.cn id=242 lang=python3
#
# [242] 有效的字母异位词
#

# @lc code=start
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s)!=len(t):
            return False
        dict={}
        for c in s:
            if (c in dict) == False:
                dict[c]=1
            else:
                dict[c]+=1
        for c in t:
            if ((c in dict) ==False) or dict[c]==0:
                return False
            dict[c]-=1
        return True

# @lc code=end

File: Storage/Code/242.有效的字母异位词.cpp
/*
 * @lc app=leetcode.cn id=242 lang=cpp
 *
 * [242] 有效的字母异位词
 */

// @lc code=start
/*
    出现unicode字符，使用unordered_map替换vector
*/
class Solution {
public:
    bool isAnagram(string s, string t) {
        int slen=s.size(), tlen=t.size();
        // 长度不等直接判否
        if(slen!=tlen){
            return false;
        }
        vector<int> svec(26, 0), tvec(26, 0);
        for(int i=0;i<slen;++i){
            ++svec[s[i]-'a'];
            ++tvec[t[i]-'a'];
        }
        for(int i=0;i<26;++i){
            if(svec[i]!=tvec[i]){
                return false;
            }
        }
        return true;
    }
};
// @lc code=end

